ledger/sandbox/src/test/suite/scala/com/digitalasset/platform/sandbox/services/completion/EmptyLedgerIT.scala

```diff
@@ -44,38 +71,118 @@ class EmptyLedgerIT
       implicitPartyAllocation = false
     )
 
-  private[this] val applicationId = getClass.getSimpleName
+  "CommandCompletionService gives sensible ledger end on an empty ledger" in {
+    val lid = ledgerId()
+    val party = "partyA"
+    val completionService = CommandCompletionServiceGrpc.stub(channel)
+    for {
+      end <- completionService.completionEnd(CompletionEndRequest(lid.unwrap))
+      completions <- completionsFromOffset(
+        completionService = completionService,
+        ledgerId = lid,
+        parties = List(party),
+        offset = end.getOffset,
+        completionTimeout = 2.seconds,
+      )
+    } yield {
+      end.getOffset.value.absolute.get shouldBe ApiOffset.begin.toHexString
+      completions shouldBe empty
+    }
+  }
+
+  "ConfigManagementService accepts a configuration when none is set" in {
+    val lid = ledgerId().unwrap
+    val completionService = CommandCompletionServiceGrpc.stub(channel)
+    for {
+      _ <- publishATimeModel(channel)
+      end <- completionService.completionEnd(CompletionEndRequest(lid))
+    } yield {
+      end.getOffset.value.absolute.get should not be ApiOffset.begin.toHexString
+    }
+  }
+
+  "CommandCompletionService can stream completions from the beginning" in {
+    val lid = ledgerId()
+    val party = "partyA"
+    val commandId = "commandId"
+
+    val submissionService = CommandSubmissionServiceGrpc.stub(channel)
+    val completionService = CommandCompletionServiceGrpc.stub(channel)
+    for {
+      end <- completionService.completionEnd(CompletionEndRequest(lid.unwrap))
+      _ <- publishATimeModel(channel)
+      _ <- uploadDarFile(darFile, channel)
+      _ <- allocateParty(party, channel)
+      completionsF = completionsFromOffset(
```

Comment:
This looks like there's flakiness potential here, if this spawned `Future` completes before the `.submit` below spawns off?

Reply:
It will always take 5 seconds to complete.

Reply:
Yeah, I guess you'd need to be majorly unlucky with the scheduling, but I don't really see that you gain anything from spawning the future earlier.

Reply:
I agree, there's flakiness potential, but hopefully not too much. If I see it flake, I'll bump the wait time.

Reply:
Well, it flaked. You were right. Reordering so we query for completions afterwards.